src/client.rs

```diff
@@ -32,16 +32,23 @@ pub struct Client {
 
 impl Client {
     /// Constructs a new `Client`.
+    #[cfg(feature = "default-tls")]
     pub fn new() -> ::Result<Client> {
         let mut client = try!(new_hyper_client());
         client.set_redirect_policy(::hyper::client::RedirectPolicy::FollowNone);
-        Ok(Client {
+        Ok(Client::with_hyper_client(client))
+    }
+
+    /// Constructs a new `Client` with the provided `::hyper::Client` without
+    /// altering any settings.
```

Comment:
Should this be promised? Specifically I'm thinking about the `RedirectPolicy`. hyper's Client has a default policy that is, well, less correct. Would it be terrible if we unconditionally disabled the redirect policy on any hyper `Client` received?